### PR TITLE
feat: add simple delegation support

### PR DESCRIPTION
**Pull request by katamarinaki**
### Summary

Following the update of the Voting contract ([Simple Delegation](https://github.com/lidofinance/aragon-apps/pull/34)), this update adds the on-chain delegation to the VotingAdapter contract, giving the TRP participants the ability to delegate their voting power. 

### Changelog

- [x] Add `assignDelegate` and `unassignDelegate` methods to the `IVoting` interface;
- [x] remove `DELEGATION_CONTRACT_ADDR` from `VotingAdapter` contract;
- [x] use `VOTING_CONTRACT_ADDR` in `delegate` method;
- [x] add new functions to the mock `Voting` contract;
- [x] update tests according to the added functionality.

### CODE REVIEW

File: contracts/VotingAdapter.vy [hunk 1]
```diff
@@ -23,6 +23,10 @@ interface IVoting:
         _supports: bool,
         _executesIfDecided_deprecated: bool,
     ): nonpayable
+    def setDelegate(
```
Comment (Psirex) [inline contracts/VotingAdapter.vy]: Semantically, it may be better to declare delegate methods in the standalone interface, as it's quite a separate aspect. Something like `IVoteDelegation` or something. Then, in the `delegate()` method, it will look like `IVoteDelegation(DELEGATION_CONTRACT_ADDR).resetDelegate()`, which is more descriptive, in my opinion.

File: tests/functional/VestingEscrow/test_delegate.py [hunk 2]
```diff
@@ -3,17 +3,20 @@
 
 def test_delegate(deployed_vesting, recipient, voting_adapter):
     data = voting_adapter.encode_delegate_calldata(recipient)
```
Comment (Psirex) [inline tests/functional/VestingEscrow/test_delegate.py:15]: Consider adding a negative test with invalid encoded data.

File: tests/functional/VotingAdapter/test_delegate.py [hunk 3]
```diff
@@ -1,7 +1,6 @@
-import brownie
 
 
 def test_delegate(voting_adapter, recipient):
     data = voting_adapter.encode_delegate_calldata(recipient)
-    with brownie.reverts("not implemented"):
-        voting_adapter.delegate(data, {"from": recipient})
+    tx = voting_adapter.delegate(data, {"from": recipient})
+    assert len(tx.events) == 1
```
Comment (Psirex) [inline tests/functional/VotingAdapter/test_delegate.py:15]: Check also, please, that event args is correct.

File: contracts/VotingAdapter.vy (hunk 1 above)
Comment (katamarinaki) [reply contracts/VotingAdapter.vy]: Agree

File: tests/functional/VestingEscrow/test_delegate.py (hunk 2 above)
Comment (dgusakov) [reply tests/functional/VestingEscrow/test_delegate.py:15]: Also consider adding test for undelegation (delegation to zero address)

Comment (katamarinaki) [reply tests/functional/VestingEscrow/test_delegate.py:15]: Addressed in https://github.com/lidofinance/lido-vesting-escrow/pull/9/commits/4246b8ddac3e00a53972baffef78ab722a00212f.

**Review by Psirex — APPROVED** [review]
